### PR TITLE
Allow negotiating plain-GET for certs

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

File: STAR/draft-ietf-acme-star.md [hunk 1]
```diff
@@ -424,7 +429,12 @@ Example directory object advertising STAR support with one day star-min-cert-val
 ## Fetching the Certificates
 {: #fetching-certificates}
 
-The certificate is fetched from the certificate endpoint, as per {{I-D.ietf-acme-acme}}, Section 7.4.2.
+The certificate is fetched from the certificate endpoint as per
+{{I-D.ietf-acme-acme}} Section 7.4.2, unless client and server have
+successfully negotiated the "unauthenticated GET" option described in
+{{certificate-get-nego}}.  In such case, the client can simply issue a GET to
+the certificate resource without authenticating herself to the server as
```
Comment (yaronf) [inline STAR/draft-ietf-acme-star.md]: herself->itself - the client is not a person.

Comment (yaronf) [reply STAR/draft-ietf-acme-star.md]: And maybe say explicitly "POST-as-GET" in addition to the reference to 7.4.2.

File: STAR/draft-ietf-acme-star.md [hunk 2]
```diff
@@ -459,6 +469,34 @@ The server MUST NOT issue any additional certificates for this order beyond its
...
+and a client to enable this service with per-Order granularity.
+
+Specifically, a server states its availability to grant unauthenticated access
+to a client's Order certificate(s) by setting the star-allow-certificate-get
+key to true in the meta field of the Directory object:
+
+- star-allow-certificate-get (optional, boolean): If this field is present and
+  set to true, the server allows GET requests to certificate URLs.
+
+A client states her will to access the issued certificate(s) via
```
Comment (yaronf) [inline STAR/draft-ietf-acme-star.md]: Similarly, her->its.

File: STAR/draft-ietf-acme-star.md [hunk 3]
```diff
@@ -459,6 +469,34 @@ The server MUST NOT issue any additional certificates for this order beyond its
...
+unauthenticated GET by adding a recurrent-certificate-get key to her Order and
+setting it to true.
+
+- recurrent-certificate-get (optional, boolean): If this field is present and
+  set to true, the client requests the server to allow unauthenticated GET to
+  the certificate(s) associated with this Order.
+
+If the server accepts the request, it MUST reflect the key in the Order.  If it
+doesn't, it SHOULD return an error response.
```
Comment (yaronf) [inline STAR/draft-ietf-acme-star.md]: I wonder about the error response (though I was in favor before): if this flag is used for reliability, not just for delegation, then fallback to POST-as-GET is reasonable. For delegation of course, this fallback would not be useful.

File: STAR/draft-ietf-acme-star.md (hunk 2 above)
Comment (thomas-fossati) [reply STAR/draft-ietf-acme-star.md]: ok, done.

File: STAR/draft-ietf-acme-star.md (hunk 1 above)
Comment (thomas-fossati) [reply STAR/draft-ietf-acme-star.md]: ok, done (both).

File: STAR/draft-ietf-acme-star.md (hunk 3 above)
Comment (thomas-fossati) [reply STAR/draft-ietf-acme-star.md]: I agree.  I'm going to drop the "SHOULD return error".

**Pull request merged**